src/sqlfluff/core/rules/functional/raw_file_slices.py

```diff
@@ -31,3 +31,25 @@ def any(self, predicate: Optional[Callable[[RawFileSlice], bool]] = None) -> boo
             if predicate is None or predicate(s):
                 return True
         return False
+
+    def select(
```

Comment:
New function `RawFileSlices.select()`, equivalent to the existing function `Segments.select()`, but for raw slices rather than segments.